provisioners/linux/bootstrap_bitbar_devicepool.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+set -e
+
+ROLE="bitbar_devicepool"
+PUPPET_REPO="https://github.com/mozilla-platform-ops/ronin_puppet.git"
+PUPPET_BRANCH="master"
+
+PUPPET_BIN='/opt/puppetlabs/bin/puppet'
+PUPPET_ENV_DIR='/etc/puppetlabs/environments'
+FACTER_BIN='/opt/puppetlabs/bin/facter'
+R10K_BIN='/opt/puppetlabs/bin/r10k'
+R10K_DIR="/etc/puppetlabs/environments/production/r10k_modules"
+ROLE_FILE='/etc/puppet_role'
+PUPPET_REPO_PATH="$PUPPET_ENV_DIR/production"
+
+FQDN=$(${FACTER_BIN} networking.fqdn)
+
+
+## FUNCTIONS
+
+function update_puppet {
```

Comment:
would it be worth sharing functions across the platforms, or merging the bootstrap scripts? This looks similar to the mojave bootstrap, and I'm guessing the linux-talos one will match closely.

Reply:
The mojave bootstrap script is oriented at bootstrapping and then cleaning up (it downloads a zip of the repo and removes on exit). There's also a run script generated for OS X. 

The bitbar script is a bit of a hybrid. It clones a repo and is designed to be a run script also. 

There are definitely parts we could pull out and reuse between them.

Part of what's nice about the scripts is that they're self contained (it's a single file to scp over) and we'd lose that if we broke stuff out (unless we wrote a simple script generator the prepended the common bits).

Reply:
That sounds reasonable. If we start duplicating to 5 or so of these scripts that are so similar, then I'd like to combine them, but even then the conditionals might be ugly.